beacon-chain/rpc/prysm/v1alpha1/validator/proposer_capella.go

```diff
@@ -2,22 +2,28 @@ package validator
 
 import (
 	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
-	"github.com/prysmaticlabs/prysm/v3/config/params"
 	"github.com/prysmaticlabs/prysm/v3/consensus-types/interfaces"
-	"github.com/prysmaticlabs/prysm/v3/time/slots"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v3/runtime/version"
 )
 
 // Sets the bls to exec data for a block.
 func (vs *Server) setBlsToExecData(blk interfaces.BeaconBlock, headState state.BeaconState) error {
-	if slots.ToEpoch(blk.Slot()) < params.BeaconConfig().CapellaForkEpoch {
+	if blk.Version() < version.Capella {
 		return nil
 	}
 
 	changes, err := vs.BLSChangesPool.BLSToExecChangesForInclusion(headState)
 	if err != nil {
+		if err := blk.Body().SetBLSToExecutionChanges([]*ethpb.SignedBLSToExecutionChange{}); err != nil {
+			log.WithError(err).Error("Could not set bls to execution data in block")
+		}
 		log.WithError(err).Error("Could not get bls to execution changes")
 	} else {
 		if err := blk.Body().SetBLSToExecutionChanges(changes); err != nil {
+			if err := blk.Body().SetBLSToExecutionChanges([]*ethpb.SignedBLSToExecutionChange{}); err != nil {
+				log.WithError(err).Error("Could not set bls to execution data in block")
```

Comment:
I think part of the requests in my feedback was to remove the error return in this function, so that we don't need to be doing this if/else in multiple places.